src/services/completions.ts

```diff
@@ -3663,71 +3664,130 @@ function getCompletionData(
-                            symbolName: firstAccessibleSymbol.name,
-                            exportName: firstAccessibleSymbol.name,
-                            fileName,
-                            moduleSpecifier,
-                        };
-                        symbolToOriginInfoMap[index] = origin;
-                    }
+                    const printer = createPrinter({
+                        removeComments: true,
+                        module: compilerOptions.module,
+                        target: compilerOptions.target,
+                        omitTrailingSemicolon: true,
+                    });
+                    const origin: SymbolOriginInfoComputedPropertyName = {
+                        kind: getNullableSymbolOriginInfoKind(SymbolOriginInfoKind.SymbolMemberNoExport) | SymbolOriginInfoKind.ComputedPropertyName,
+                        symbolName: printer.printNode(EmitHint.Unspecified, node, contextToken.getSourceFile()),
+                    };
+                    symbolToOriginInfoMap[index] = origin;
                 }
             }
-            else if (preferences.includeCompletionsWithInsertText) {
-                if (firstAccessibleSymbolId && seenPropertySymbols.has(firstAccessibleSymbolId)) {
-                    return;
+            else {
+                const fileName = isExternalModuleNameRelative(stripQuotes(moduleSymbol.name)) ? getSourceFileOfModule(moduleSymbol)?.fileName : undefined;
+                const { moduleSpecifier } = (importSpecifierResolver ||= codefix.createImportSpecifierResolver(sourceFile, program, host, preferences)).getModuleSpecifierForBestExportInfo(
+                    [{
+                        exportKind: ExportKind.Named,
+                        moduleFileName: fileName,
+                        isFromPackageJson: false,
+                        moduleSymbol,
+                        symbol: firstAccessibleSymbol,
+                        targetFlags: skipAlias(firstAccessibleSymbol, typeChecker).flags,
+                    }],
+                    position,
+                    isValidTypeOnlyAliasUseSite(location),
+                ) || {};
+
+                if (moduleSpecifier) {
+                    const origin: SymbolOriginInfoResolvedExport = {
+                        kind: getNullableSymbolOriginInfoKind(SymbolOriginInfoKind.SymbolMemberExport),
+                        moduleSymbol,
+                        isDefaultExport: false,
+                        symbolName: firstAccessibleSymbol.name,
+                        exportName: firstAccessibleSymbol.name,
+                        fileName,
+                        moduleSpecifier,
+                    };
+                    symbolToOriginInfoMap[index] = origin;
                 }
-                addSymbolOriginInfo(symbol);
-                addSymbolSortInfo(symbol);
-                symbols.push(symbol);
             }
         }
-        else {
+
+        function addLiteralSymbol() {

```

Comment:
Not very important, but `addLiteralSymbol` makes me think we're adding a symbol for a literal, maybe simply `addSymbol` would be better.